### PR TITLE
♻️ Convert to using modern gem defaults

### PR DESCRIPTION
**Pull request by nevans**
This is roughly the result of running `bundle gem staccato`, merging and configuring that result with the existing project, and adding a few binstubs.  It adds rubocop, GitHub Actions, and rbs.

The required ruby version is raised from 2.0 to 2.6, so that a (mostly) up-to-date rubocop can be run—all 2.x versions of ruby are EOL and this simplifies requirements for running rubocop and other CI tools.  This also simplifies any future PRs, by allowing them to take advantage of more modern ruby features.

In addition to the default rubocop config provided by the `bundle gem` template, I setup the baseline rubocop config to inherit from `standard` gem.  Contrary to the "standard" propaganda 😉, it's okay to inherit from, disable, and override the "standard" config.  Its config is better than the rubocop defaults, IMO, and a better fit for the pre-existing style used in this project.  Where this project significantly deviates from the "standard" config, I used my personal judgement about which rules to disable or configure in `rubocop.yml` and which ones to generate into `rubocop_todo.yml`... _and which ones to autocorrect_.

The Travis config was replaced with GitHub Actions, and it runs on every supported version of ruby.

The rbs file was simply generated by typeprof, and I didn't audit the generated rbs for correctness or completeness.

The Code of Conduct was updated to the more recent version provided by `bundle gem`.

----

_Note: If you would prefer to continue supporting ruby 2.0-2.5, or turn off all rubocop style rules by default or use some other baseline rubocop configuration, or have any other aspects of this that you'd like me to change, let me know.  I'm creating this as simply a draft PR, to initiate that discussion._

_Additionally, because I don't have permissions to run workflows in this repository, I've created a mirror PR in my own repo, and you can see the workflow results here: https://github.com/nevans/staccato/pull/1/checks._

### CODE REVIEW

File: sig/staccato.rbs [hunk 1]
```diff
@@ -0,0 +1,367 @@
+module Staccato
```
Comment (tpitale) [inline sig/staccato.rbs:1]: Is this file required for rubocop or something?

File: bin/rake [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
```
Comment (tpitale) [inline bin/rake:1]: What's the current deal with binstubs? Is this the standard in ruby land at this point vs `bundle exec rake`?

Comment (nevans) [reply bin/rake:1]: I suppose it's somewhere between a partial standard and merely a convenience.  But `bundle gem $gemname` only creates `bin/setup` and `bin/console`.

I _personally_ find it slightly nicer to use the binstubs, so I like to create them for rake and rspec and anything else I'm running frequently.  I don't know how widespread or "standard" that is though, but rails does it, bundler makes it easy to do, and I do sometimes see it in other projects.

File: sig/staccato.rbs (hunk 1 above)
Comment (nevans) [reply sig/staccato.rbs:1]: No, it isn't.  It should be usable by the `steep`, `rbs`, and `typeprof` gems, but I confess that I don't have much experience with rbs at the moment.

`bundle gem staccato` created the file, and I used `typeprof` to automatically create its contents.  IMO, if

Comment (nevans) [reply sig/staccato.rbs:1]: IMO, it would be much nicer if bundler's project template also created some rake tasks for both _checking_ types and regenerating the RBS.  Since I don't have any type-checking in the CI build, I'm inclined to just delete this file: otherwise it will get out of sync with the code.  What do you think?

File: bin/rake (hunk 2 above)
Comment (tpitale) [reply bin/rake:1]: I'm cool with binstubs, that's fine by me.

File: sig/staccato.rbs (hunk 1 above)
Comment (tpitale) [reply sig/staccato.rbs:1]: I do like the idea of the type checking, but if you're okay with it, let's punt on this for this PR. Thanks!

**Comment by nevans** [comment]
@tpitale thanks for taking the time to give it a once over.

File: sig/staccato.rbs (hunk 1 above)
Comment (nevans) [reply sig/staccato.rbs:1]: 👍🏻 I'll delete this file.
